app/src/main/java/org/schabi/newpipe/views/ScrollableTabLayout.java

```diff
@@ -0,0 +1,91 @@
+package org.schabi.newpipe.views;
+
+import android.content.Context;
+import android.os.Build;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewTreeObserver.OnGlobalLayoutListener;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.material.tabs.TabLayout;
+import com.google.android.material.tabs.TabLayout.Tab;
+
+/**
+ * A TabLayout that is scrollable when tabs exceed its width.
+ * Hides when there are less than 2 tabs.
+ */
+public class ScrollableTabLayout extends TabLayout {
+    private static final String TAG = ScrollableTabLayout.class.getSimpleName();
+
+    public ScrollableTabLayout(Context context) {
+        super(context);
+    }
+
+    public ScrollableTabLayout(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public ScrollableTabLayout(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+    @Override
+    protected void onAttachedToWindow() {
+        super.onAttachedToWindow();
+
+        setTabMode(TabLayout.MODE_FIXED);
+        resetMode();
+    }
+
+    @Override
+    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
+        super.onSizeChanged(w, h, oldw, oldh);
+
+        resetMode();
+    }
+
+    @Override
+    public void addTab(@NonNull Tab tab, int position, boolean setSelected) {
+        super.addTab(tab, position, setSelected);
+
+        resetMode();
+    }
+
+    @Override
+    public void removeTabAt(int position) {
+        super.removeTabAt(position);
+
+        resetMode();
+    }
+
+    private void resetMode() {
+        if (getTabCount() < 2) {
+            setVisibility(View.GONE);
+            return;
+        } else {
+            setVisibility(View.VISIBLE);
+        }
+
+        if (getTabCount() == 0 || getTabAt(0).view == null) return;
+        setTabMode(TabLayout.MODE_FIXED);
+
+        int layoutWidth = getWidth();
+        int minimumWidth = ((View) getTabAt(0).view).getMinimumWidth();
+        if (minimumWidth * getTabCount() > layoutWidth) {
+            setTabMode(TabLayout.MODE_SCROLLABLE);
+            return;
+        }
+
+        int actualWidth = 0;
+        for (int i = 0; i < getTabCount(); ++i) {
+            if (getTabAt(i).view == null) return;
+            actualWidth += ((View) getTabAt(i).view).getWidth();
+            if (actualWidth > layoutWidth) {
+                setTabMode(TabLayout.MODE_SCROLLABLE);
+                return;
+            }
+        }
+    }
```

Comment:
Could you add some comments to this function? For example explaining why `getTabAt(0).view` could be `null` and why you check for width twice (I assume you first do a fast check to improve performance?).

Reply:
`getTabAt()` has a `@Nullable` annotation. I'll make this function a little cleaner and account for possible null tabs differently.

Reply:
Nevermind, it returns `null` if index exceeds `getTabCount()`.

There were no checks for whether `Tab.view` is null in some `Tab`'s public methods modifying `view` so I think I can assume that it's not null?

Reply:
I added a few comments throughout the class and simplified the calculation of tabs' width. I also fixed my mistake where width was calculated based only on minimum width requested by tab even if it's contents were wider.